Add docs for monitoring private networks via synthetics service (#2583)
elastic/observability-docs: docs/en/observability/synthetics-private-location.asciidoc

[[synthetics-private-location]]

> = Monitor a private endpoint

Reviewer: ```suggestion
= Monitor resources on private networks
```

> To monitor resources on private networks use {{agent}} to create a Private Location, or allow access to your private endpoints to the Synthetics Service.

Reviewer: A couple thoughts:

* I would order the options the same way they are organized on the page. It looks like you put the private endpoint section before creating a private location.
* This sentence was confusing to me:

  >allow access to your private endpoints to the Synthetics Service

  but feel free to adjust my suggestion changed the meaning.

```suggestion
To monitor resources on private networks you can either:

* Allow the Synthetics service to access your private endpoints.
* Use {agent} to create a Private Location.

```

Author: Much cleaner, yes :)

Private Locations via Elastic Agent require only outbound connections from your network, while allowing access to the Synthetics Service requires allowing

> inbound access, thus posing an additional risk that users must assess.

Reviewer: This gives no reason why you _would_ choose to allow the Synthetics service to access a private endpoint, it only warns you _against_ doing it. When _should_ you use it?

>thus posing an additional risk that users must assess

Also, who does `users` refer to here?

Author: Good point, you'd do it because you don't want to run infrastructure yourself, will add that

[discrete]
[[monitor-via-access-control]]

> = Allow the synthetics service access to your private network

Reviewer: Choose one style:

* Synthetics Service
* synthetics service
* Synthetics service

```suggestion
= Allow the Synthetics service access to your private network
```

Author: I think it should be "Synthetics Service". WDYT? If you're OK with that I'll add it here. Should it become a variable?

Reviewer 2: The _proper_name is `Elastic's global managed infrastructure` like we use [elsewhere](https://www.elastic.co/guide/en/observability/master/synthetics-private-location.html#synthetics-private-location-next), should probably use that for consistency

> To grant access to a private endpoint to our Synthetics Service use one or both of IP address filtering and HTTP authentication.

Reviewer: ```suggestion
To give the Synthetics Service access to a private endpoint, use IP address filtering, HTTP authentication, or both.
```

> To grant access via IP, use the list of egress IPs maintaned at https://manifest.synthetics.elastic-cloud.com/v1/ip-ranges.json[this URL].

Reviewer: Use descriptive link text.

```suggestion
To grant access via IP, use https://manifest.synthetics.elastic-cloud.com/v1/ip-ranges.json[this list of egress IPs].
```

> Please note that the addresses and locations on this list may be updated from time to time, so automating the updating of your
> filtering rules is recommended. Note that IP filtering alone will allow all users of our synthetics service access to your endpoints, if this
> is a concern consider adding additional protection via user/password authentication via a proxy like nginx.

Reviewer: ```suggestion
The addresses and locations on this list may change, so automating updates to
filtering rules is recommended. IP filtering alone will allow all users of our synthetics service access to your endpoints, if this
is a concern consider adding additional protection via user/password authentication via a proxy like nginx.
```

[[synthetics-private-location]]
= Monitor resources on private networks

> To monitor resources on private networks you can either:
> 
> * Allow the Synthetics service to access your private endpoints.
> * Use {agent} to create a Private Location.
> 
> Private Locations via Elastic Agent require only outbound connections from your network, while allowing access to the Synthetics Service requires allowing
> inbound access, thus posing an additional risk that users must assess.
> 
> [discrete]
> [[monitor-via-access-control]]
> = Allow the synthetics service access to your private network
> 
> To give the Synthetics Service access to a private endpoint, use IP address filtering, HTTP authentication, or both.
> 
> To grant access via IP, use https://manifest.synthetics.elastic-cloud.com/v1/ip-ranges.json[this list of egress IPs].
> The addresses and locations on this list may change, so automating updates to
> filtering rules is recommended. IP filtering alone will allow all users of our synthetics service access to your endpoints, if this
> is a concern consider adding additional protection via user/password authentication via a proxy like nginx.

Reviewer: >The _proper_name is Elastic's global managed infrastructure like we use elsewhere, should probably use that for consistency

@paulb-elastic @andrewvc does "Elastic's global managed infrastructure" make sense in all these contexts? 

```suggestion
To monitor resources on private networks you can either:

* Allow Elastic's global managed infrastructure to access your private endpoints.
* Use {agent} to create a Private Location.

Private Locations via Elastic Agent require only outbound connections from your network, while allowing Elastic's global managed infrastructure to access a private endpoint requires
inbound access, thus posing an additional risk that users must assess.

[discrete]
[[monitor-via-access-control]]
= Allow access to your private network

To give Elastic's global managed infrastructure access to a private endpoint, use IP address filtering, HTTP authentication, or both.

To grant access via IP, use https://manifest.synthetics.elastic-cloud.com/v1/ip-ranges.json[this list of egress IPs].
The addresses and locations on this list may change, so automating updates to
filtering rules is recommended. IP filtering alone will allow all users of Elastic's global managed infrastructure access to your endpoints, if this
is a concern consider adding additional protection via user/password authentication via a proxy like nginx.
```

Reviewer 2: > ...does "Elastic's global managed infrastructure" make sense in all these contexts?

I do think that it is wordy, but it still makes sense to me when reading it.  My worry is introducing other terms (we've had this in the past).  I know internally we sometimes distinguish them as _Public_ and _Private_ locations, but haven't used that externally as far as I can remember.

Reviewer: >I do think that it is wordy, but it still makes sense to me when reading it. My worry is introducing other terms (we've had this in the past).

I agree. For the purposes of this PR I think it's ok to use the wordy phrase to keep docs consistent. Then we can revisit   this terminology separately, and if we decide to change to something like "Synthetics Service" we can make sure that change is applied across all docs. What do you think?

Author: I'm fine with the wordier phrase
